### PR TITLE
docs(input-number, input-text): remove unsupported boolean type from type prop doc

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #6386

## Summary

In the future, we will be removing the support of the `boolean` type for the `icon` prop on `input-number` and `input-text` as these don't have a `default` recommended icon for an unspecified icon attribute and result in unwanted spacing preceding the `input`. 

For now though, since it is potentially a minor breaking change that could possibly be in use somewhere, we removed the `boolean` scenario from the documentation and marked it with a newly introduced custom `@futureBreaking` tag.

### CODE REVIEW

File: src/components/input-number/input-number.tsx [hunk 1]
```diff
@@ -808,7 +808,7 @@ export class InputNumber
     };
 
     const sanitizedValue = sanitizeNumberString(
-        // no need to delocalize a string that ia already in latn numerals
+      // no need to delocalize a string that ia already in latn numerals
```
Comment (jcfranco) [inline src/components/input-number/input-number.tsx:811]: Was formatted by the linter?

Typo (unrelated): `ia` -> `is`

File: src/components/input-number/input-number.scss [hunk 2]
```diff
@@ -137,15 +137,15 @@
 
 // position placeholder/value text in relation to icons
 
-:host([icon][scale="s"]) input {
+:host([icon]:not([icon=""])[scale="s"]) input {
```
Comment (jcfranco) [inline src/components/input-number/input-number.scss:140]: If the doc and type no longer allows boolean, I think we can drop `:not([icon=""])` and assume `icon` has a value.

File: src/components/input-number/input-number.tsx (hunk 1 above)
Comment (Elijbet) [reply src/components/input-number/input-number.tsx:811]: Yes it is.

File: src/components/input-number/input-number.tsx [hunk 3]
```diff
@@ -131,9 +131,9 @@ export class InputNumber
   @Prop({ reflect: true }) hidden = false;
 
   /**
-   * When `true`, shows a default recommended icon. Alternatively, pass a Calcite UI Icon name to display a specific icon.
+   * Pass a Calcite UI Icon name to display a specific icon.
```
Comment (geospatialem) [inline src/components/input-number/input-number.tsx:134]: WDYT about the following (in line with other components such as `action` and `chip`:

> Specifies an icon to display.

File: src/components/input-text/input-text.tsx [hunk 4]
```diff
@@ -109,9 +109,9 @@ export class InputText
   @Prop({ reflect: true }) hidden = false;
 
   /**
-   * When `true`, shows a default recommended icon. Alternatively, pass a Calcite UI Icon name to display a specific icon.
+   * Pass a Calcite UI Icon name to display a specific icon.
```
Comment (geospatialem) [inline src/components/input-text/input-text.tsx:112]: Similar to above:

> Specifies an icon to display.

**Review by jcfranco — APPROVED** [review]
💥

**Pull request merged**